deidentification/utils/get_tags_diff.py

```diff
@@ -0,0 +1,90 @@
+        outpath (str): path of the diff_file
+    """
+    print(f"Check difference between :\n{path_raw} and\n{path_ano}\n")
+    dcm_ref = pydicom.read_file(path_raw)
+    dcm_current = pydicom.read_file(path_ano)
+    diff = difflib.ndiff(str(dcm_ref).splitlines(keepends=True),
+                         str(dcm_current).splitlines(keepends=True))
+    
+    diff_from_dcm1 = []
+    diff_from_dcm_2 = []
+    global_modif = []
+    current_line_diff = None
+    previous_tag = None
+    previous_sign =  None
+
+    for i in diff:
+        if "(" not in i or ")" not in i:
+            continue
+        tag = i[i.index('('): i.index(')')+1]
+        txt = ""
+
+        # tag present in only one dcm
+        if tag != previous_tag:
+            if i[0] == "-":
+                txt = "Dcm1 only:"
+                current_line_diff = diff_from_dcm1
+            elif i[0] == "+":
+                txt = "Dcm2 only:"
+                current_line_diff = diff_from_dcm_2
+            line = txt + i[1:]
+        # between 2 diff lines : same tag + remove from first line and add to second -> value modification
+        elif tag == previous_tag and i[0] != previous_sign:
+            line = ""
+            txt = f"Modification (dcm1 -> dcm2): {tag}"
+            previous_elem = current_line_diff.pop(-1)
+            # check tag name
+            previous_name = previous_elem[previous_elem.index(')') + 1: previous_elem.index(re.search(' [A-Z]{2}:', previous_elem).group())].replace("  ", "")
+            current_name = i[i.index(')') + 1: i.index(re.search(' [A-Z]{2}:', i).group())].replace("  ", "")
+            if current_name != previous_name:
+                line += txt + f'\tName modification\t {previous_name} ---> {current_name}\n'
+
+            # check tag value
+            previous_value = previous_elem.split(': ')[-1].replace('\n', '')
+            current_value = i.split(': ')[-1].replace('\n', '') 
+            if previous_value != current_value:
+                line += txt + "\tValue modification\t" + previous_value + " -----> " + current_value + '\n'
+            current_line_diff = global_modif
+
+        if txt != "":
+            previous_tag = tag
+            previous_sign = i[0]
+            current_line_diff.append(line)
+
+    if not outpath:
+        outpath = os.path.join(os.path.dirname(path_raw), "diff.txt")
+    with open(outpath, "w") as f:
+        f.write(f'dcm1 = {path_raw}\n')
+        f.write(f'dcm2 = {path_ano}\n\n\n')
+        f.writelines(diff_from_dcm1)
+        f.write('\n\n')
+        f.writelines(diff_from_dcm_2)
+        f.write('\n\n')
+        f.writelines(global_modif)
+    
+    print("Done\n")
+    print("Output file : " + os.path.join(os.path.dirname(path_raw), "diff.txt"))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-dcm1', required=True)
+    parser.add_argument('-dcm2', required=True)
+    parser.add_argument('-outpath', required=False)
```

Comment:
Command lines parameters are usually declared as:

- short argument with one dash (`-h`, `-v`, ...)
- long argument with two dashes (`--help`, `--verbose`, ...)

If we want to follow this rule, we may have `--dcm1` and `--outpath`. You can also declare a long and a short parameter name as
```python
parser.add_argument('-o', '--outpath', required=False)
```